### PR TITLE
replace new instead of std::make_unique or std::make_shared

### PR DESCRIPTION
**Pull request by ghost**
1. replaced new instead of std::make_unique or std::make_shared 
2. replace some NULLs instead of nullptr
3. replace condition == false instead of !condition

### CODE REVIEW

**Comment by PragmaTwice** [comment]
Hi, thanks for your contribution.

Since there is no `std::make_unique` in c++11, you can use `Util::MakeUnique` in `util.h` as an alternative.

**Comment by git-hulk** [comment]
Hello @Furud0Er1ka, I saw many places are only formatted without changing the logic, I think we can rollback those format codes to keep this PR consistent with what's title described, it also helps the code review easier.

File: src/util.h [hunk 1]
```diff
@@ -71,8 +76,14 @@ uint64_t GetTimeStampUS(void);
...
 }
 
-}  // namespace Util
+// define std::make_shared in c++14
+// refer to https://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared
+template <typename T, typename... Args>
+inline std::shared_ptr<T> MakeShared(Args &&...args) {
+  return std::shared_ptr<T>(new T(std::forward<Args>(args)...));
+}
+} // namespace Util
```
Comment (PragmaTwice) [inline src/util.h]: `std::make_shared` is in c++11, not c++14, so I think we do not need it.

File: src/util.h [hunk 2]
```diff
@@ -71,8 +76,14 @@ uint64_t GetTimeStampUS(void);
 // define std::make_unique in c++14
 // refer to https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique
 template <typename T, typename... Args>
-std::unique_ptr<T> MakeUnique(Args&& ... args) {
+inline std::unique_ptr<T> MakeUnique(Args &&...args) {
```
Comment (PragmaTwice) [inline src/util.h]: This `inline` is useless. A function template does already imply `inline`.

**Comment by PragmaTwice** [comment]
Hi, as mentioned by @git-hulk, I think you could minimize your PR, and you can check here to see why cpplint failed: https://github.com/apache/incubator-kvrocks/runs/7902217149?check_suite_focus=true

**Comment by ghost** [comment]
> Hi, as mentioned by @git-hulk, I think you could minimize your PR, and you can check here to see why cpplint failed: https://github.com/apache/incubator-kvrocks/runs/7902217149?check_suite_focus=true

thanks for your review! I have removed all unnecessary changes.

File: src/util.h [hunk 3]
```diff
@@ -74,5 +74,8 @@ template <typename T, typename... Args>
 std::unique_ptr<T> MakeUnique(Args&& ... args) {
   return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
 }
-
+template <typename T, typename... Args>
```
Comment (git-hulk) [inline src/util.h:78]:
```suggestion

template <typename T, typename... Args>
```

**Comment by git-hulk** [comment]
cool, thanks

**Review by git-hulk — DISMISSED** [review]

File: src/util.h [hunk 4]
```diff
@@ -75,4 +75,8 @@ std::unique_ptr<T> MakeUnique(Args&& ... args) {
   return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
 }
 
+template <typename T, typename... Args>
+std::shared_ptr<T> MakeShared(Args&& ... args) {
+  return std::shared_ptr<T>(new T(std::forward<Args>(args)...));
```
Comment (PragmaTwice) [inline src/util.h]:
```suggestion
  return std::make_shared<T>(std::forward<Args>(args)...);
```

**Review by PragmaTwice — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Comment by PragmaTwice** [comment]
Thanks for your contribution again! Merging...

**Pull request merged**